calc.go

```diff
@@ -1,22 +1,25 @@
 package pokemonbattlelib
 
-func calcMoveDamage(weather Weather, user, receiver *Pokemon, move *Move) (damage uint) {
-	weatherMod := 1.0
-	if rain, sun := weather == WeatherRain, weather == WeatherHarshSunlight; (rain && move.Type() == TypeWater) || (sun && move.Type() == TypeFire) {
-		weatherMod = 1.5
+func calcMoveDamage(b *Battle, user, receiver *Pokemon, move *Move) (damage float64) {
```

Comment:
Don't change this function signature, it intentionally doesn't take a `*Battle` and returns a `uint`